src/components/simulation/tray-image.tsx

```diff
@@ -11,38 +11,57 @@ import "./tray-image.scss";
 // corner. This ends up being about 5 pixels for all tray objects. Technically it could be computed
 // by hand for ALL tray objects, but subtracting 5 pixels works pretty well in practice.
 const kOutlineOffset = 5;
+const kNonTrayPreviewHeight = 48;
 
 interface IProps {
   trayObject: TrayObject;
   onTrayObjectSelect: (type: TrayType) => void;
   traySelectionType?: TrayType;
+  trayWrapper: HTMLDivElement | null;
 }
 
 export const TrayImage: React.FC<IProps> = (props) => {
-  const { trayObject, onTrayObjectSelect, traySelectionType } = props;
+  const { trayObject, onTrayObjectSelect, traySelectionType, trayWrapper } = props;
   const TrayObjectImage = trayObject.image;
 
-  const [{isDragging, dragSourcePosition}, drag ] = useDrag({
+  const [{isDragging, dragSourcePosition, dragPosition}, drag ] = useDrag({
     item: { type: trayObject.type, trayIndex: trayObject.trayIndex, dragImage: trayObject.dragImage,
             left: trayObject.left, top: trayObject.top },
     collect: monitor => ({
       isDragging: !!monitor.isDragging(),
-      dragSourcePosition: monitor.getSourceClientOffset()
+      dragPosition: monitor.getClientOffset(),
+      dragSourcePosition: monitor.getSourceClientOffset(),
     }),
   });
 
   const PreviewImage = () => {
     const {display, item} = usePreview();
-    if (!display || !dragSourcePosition) {
+    if (!display || !dragSourcePosition || !dragPosition) {
       return null;
     }
     const boundingBoxDeltaX = (trayObject.boundingBoxWidth - trayObject.width) / 2;
     const boundingBoxDeltaY = (trayObject.boundingBoxHeight - trayObject.height) / 2;
-    const previewStyle = {
-      left: dragSourcePosition.x + boundingBoxDeltaX - kOutlineOffset,
-      top: dragSourcePosition.y + boundingBoxDeltaY - kOutlineOffset,
-      transform: `rotate(${trayObject.rotation}deg)`
-    };
+    let previewStyle: React.CSSProperties;
+    const isLeaf = draggableLeafTypes.includes(trayObject.type as any);
+
+    const trayRect = trayWrapper?.getBoundingClientRect();
+    const overTray = trayRect && (dragPosition.x >= trayRect.x && dragPosition.x <= (trayRect.x + trayRect.width) &&
+                     dragPosition.y >= trayRect.y && dragPosition.y <= (trayRect.x + trayRect.height));
```

Comment:
`getBoundingClientRect()` also returns `left`, `right`, `bottom`, `top`, so you don't technically need to do the calculation yourself.